### PR TITLE
mark txn-gc-RFC as in-progress and some boring renames

### PR DESCRIPTION
**Pull request by tbg**
[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3116)

### CODE REVIEW

**Comment by bdarnell** [comment]
I don't like "span" as a name here. I guess `SequenceSpan` is a span in the sense of being a contiguous chunk of the keyspace, but we generally use "span" to refer to the pair of keys that define the endpoints of a span.

**Comment by tschottdorf** [comment]
Bring a better suggestion and I'll rename accordingly.

**Comment by bdarnell** [comment]
I think I prefer SequenceCache to SequenceSpan. I can't think of anything else.

**Comment by tschottdorf** [comment]
ok, `SequenceCache` it is.

File: storage/replica_test.go [hunk 1]
```diff
@@ -2048,7 +2048,7 @@ func TestPushTxnBadKey(t *testing.T) {
 func TestPushTxnAlreadyCommittedOrAborted(t *testing.T) {
 	defer leaktest.AfterTest(t)
 	// This test simulates running into an open intent and resolving it
-	// using the Txn table. If we auto-gc'ed entries here, the entry would
+	// using the Txn span. If we auto-gc'ed entries here, the entry would
```
Comment (bdarnell) [inline storage/replica_test.go]: s/span/cache/

File: storage/sequence_cache.go [hunk 2]
```diff
@@ -81,29 +81,28 @@ func (rc *ResponseCache) GetSequence(e engine.Engine, family []byte) (int64, err
...
+		keys.SequenceCacheKey(rc.rangeID, roachpb.KeyMin))
 	end := engine.MVCCEncodeKey(
-		keys.ResponseCacheKey(rc.rangeID, roachpb.KeyMax))
+		keys.SequenceCacheKey(rc.rangeID, roachpb.KeyMax))
 
 	return e.Iterate(start, end, func(kv engine.MVCCKeyValue) (bool, error) {
 		// Decode the key into a cmd, skipping on error. Otherwise,
-		// write it to the corresponding key in the new cache.
-		family, err := rc.decodeResponseCacheKey(kv.Key)
+		// write it to the corresponding key in the new span.
```
Comment (bdarnell) [inline storage/sequence_cache.go]: s/span/cache/

File: storage/sequence_cache.go [hunk 3]
```diff
@@ -113,31 +112,31 @@ func (rc *ResponseCache) CopyInto(e engine.Engine, destRangeID roachpb.RangeID)
 	})
 }
 
-// CopyFrom copies all the cached results from the originRangeID
-// response cache into this one. Note that the cache will not be
+// CopyFrom copies all the persisted results from the originRangeID
+// sequence cache into this one. Note that the cache will not be
 // locked while copying is in progress. Failures decoding individual
-// cache entries return an error. The copy is done directly using the
+// span entries return an error. The copy is done directly using the
```
Comment (bdarnell) [inline storage/sequence_cache.go]: s/span/cache/

File: storage/sequence_cache.go [hunk 4]
```diff
@@ -113,31 +112,31 @@ func (rc *ResponseCache) CopyInto(e engine.Engine, destRangeID roachpb.RangeID)
...
+		keys.SequenceCacheKey(originRangeID, roachpb.KeyMin))
 	end := engine.MVCCEncodeKey(
-		keys.ResponseCacheKey(originRangeID, roachpb.KeyMax))
+		keys.SequenceCacheKey(originRangeID, roachpb.KeyMax))
 
 	return e.Iterate(start, end, func(kv engine.MVCCKeyValue) (bool, error) {
 		// Decode the key into a cmd, skipping on error. Otherwise,
-		// write it to the corresponding key in the new cache.
-		family, err := rc.decodeResponseCacheKey(kv.Key)
+		// write it to the corresponding key in the new span.
```
Comment (bdarnell) [inline storage/sequence_cache.go]: s/span/cache/

File: storage/sequence_cache.go [hunk 5]
```diff
@@ -157,16 +156,16 @@ func (rc *ResponseCache) PutSequence(e engine.Engine, family []byte, sequence in
...
 	v.SetInt(sequence)
 	return engine.MVCCPut(e, nil /* ms */, key, roachpb.ZeroTimestamp, v, nil /* txn */)
 }
 
 // Responses with write-too-old, write-intent and not leader errors
-// are retried on the server, and so are not recorded in the response
-// cache in the hopes of retrying to a successful outcome.
-func (rc *ResponseCache) shouldCacheError(err error) bool {
+// are retried on the server, and so are not recorded in the sequence
+// span in the hopes of retrying to a successful outcome.
```
Comment (bdarnell) [inline storage/sequence_cache.go]: s/span/cache/

**Comment by bdarnell** [comment]
LGTM

File: storage/sequence_cache.go (hunk 2 above)
Comment (tschottdorf) [inline storage/sequence_cache.go]: done.

File: storage/replica_test.go (hunk 1 above)
Comment (tschottdorf) [inline storage/replica_test.go]: done.

File: storage/sequence_cache.go (hunk 3 above)
Comment (tschottdorf) [inline storage/sequence_cache.go]: done.

File: storage/sequence_cache.go (hunk 4 above)
Comment (tschottdorf) [inline storage/sequence_cache.go]: done.

File: storage/sequence_cache.go (hunk 5 above)
Comment (tschottdorf) [inline storage/sequence_cache.go]: done.

**Pull request merged**